Refactor Dependencies::process() with exceptions

siwinski opened the pull request.

siwinski committed:
Refactor Dependencies::process() with exceptions

siwinski commented on src/Dependencies.php:
```diff
@@ -94,26 +94,30 @@ private function __construct()
+                        requireFile($firstExistsDependency);
+                        $loadedDependency = $firstExistsDependency;
                         break;
+                    } catch (\RuntimeException $e) {
```
For the change here, I was trying to make this block of code use the same conditionals logic as `Fedora\Autoloader::requireFile()` without copying-and-pasting the conditionals.  This does have the overhead of an Exception object creation (from `Fedora\Autoloader::requireFile()`) and the catch though.  Perhaps a copy-and-paste of the conditionals here would be better or a common function (something like `Fedora\Autoloader::fileLoadable()`?) that this code block and `Fedora\Autoloader::requireFile()` could both use?

remicollet commented on src/Dependencies.php:
```diff
@@ -94,26 +94,30 @@ private function __construct()
         foreach ($dependencies as $dependency) {
             if (is_array($dependency)) {
-                $dependencyToLoad = null;
+                $loadedDependency = null;
```
A boolean can be enough ?

siwinski replied:
Yes!  Updated.

remicollet commented on src/Dependencies.php:
```diff
@@ -94,26 +94,30 @@ private function __construct()
-                    requireFile($dependencyToLoad);
+                if ($required && empty($loadedDependency)) {
+                    throw new \RuntimeException(sprintf(
+                      'Files not found: "%s"',
```
for consistency with other exception use single quote around each name.

siwinski replied:
Updated.

remicollet reviewed (approved):
no problem for me, feel free to merge for 0.2.0 is you want

siwinski committed:
[Dependencies] Process updates

- Boolean to test if "first found" dependency loaded instead of full dependency string
- Replace double-quotes with single-quotes in "First found" dependency thrown exception
- Throw exception when single (not "first found") dependency is required and not loaded

siwinski commented:
Updated

On Wed, Oct 26, 2016 at 1:51 AM, Remi Collet notifications@github.com
wrote:

> _@remicollet_ approved this pull request.
> 
> ## no problem for me, feel free to merge for 0.2.0 is you want
> 
> In src/Dependencies.php
> https://github.com/php-fedora/autoloader/pull/9#pullrequestreview-5790592
> :
> 
> > ```
> >   */
> >  protected static function process(array $dependencies, $required)
> >  {
> >      foreach ($dependencies as $dependency) {
> >          if (is_array($dependency)) {
> > ```
> > -                $dependencyToLoad = null;
> > -                $loadedDependency = null;
> 
> ## A boolean can be enough ?
> 
> In src/Dependencies.php
> https://github.com/php-fedora/autoloader/pull/9#pullrequestreview-5790592
> :
> 
> > ```
> >                  }
> >              }
> > ```
> -                if ($required || isset($dependencyToLoad)) {
> -                    if (!isset($dependencyToLoad)) {
> -                        $dependencyToLoad = end($dependency);
> 
> ## \-                    }
> -                    requireFile($dependencyToLoad);
> -                if ($required && empty($loadedDependency)) {
> -                    throw new \RuntimeException(sprintf(
> -                      'Files not found: "%s"',
> 
> for consistency with other exception use single quote around each name.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/php-fedora/autoloader/pull/9#pullrequestreview-5790592,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AA_HKNcLny1ROFKeCOUtLT0Nt7x8HsSQks5q3up1gaJpZM4Kgtjd
> .
